File: packages/gbnf-experiment/data/20260909T124157Z_4ee67ba2/ported_implementation/tests/unit/utils/errors/input-parse-error.test.ts
import assert from 'node:assert/strict';
import { describe, test } from 'node:test';
import {
  INPUT_PARSER_ERROR_HEADER_MESSAGE,
  InputParseError,
} from '../../../../src/utils/errors/input-parse-error.ts';

describe('input parse error', () => {
  test('it renders a message', () => {
    const input = 'some input';
    const err = new InputParseError(input, 1);
    assert.equal(
      err.message,
      [INPUT_PARSER_ERROR_HEADER_MESSAGE, '', input, ' ^'].join('\n'),
    );
  });

  test('it renders a message for code point', () => {
    const err = new InputParseError('a', 0);
    assert.equal(err.message, [INPUT_PARSER_ERROR_HEADER_MESSAGE, '', 'a', '^'].join('\n'));
  });

  test('it renders a message for code points', () => {
    const err = new InputParseError('abcd', 2);
    assert.equal(
      err.message,
      [INPUT_PARSER_ERROR_HEADER_MESSAGE, '', 'abcd', '  ^'].join('\n'),
    );
  });
});

File: proposed-projects/round-trip-experiment/reverse/20260909T002206Z_273cf6b9/20260910T042804Z_e72f6c04/ported_implementation/gbnf/utils/errors/input_parse_error_test.py
from .input_parse_error import INPUT_PARSER_ERROR_HEADER_MESSAGE, InputParseError


def test_it_renders_a_message():
    input = "some input"
    pos = 1
    err = InputParseError(input, pos)
    assert err.args[0] == "\n".join(
        [
            INPUT_PARSER_ERROR_HEADER_MESSAGE,
            "",
            input,
            " ^",
        ]
    )


def test_it_renders_a_message_for_a_code_point():
    err = InputParseError("a", 0)
    assert err.args[0] == "\n".join(
        [
            INPUT_PARSER_ERROR_HEADER_MESSAGE,
            "",
            "a",
            "^",
        ]
    )


def test_it_renders_a_message_for_code_points():
    err = InputParseError("abcd", 2)
    assert err.args[0] == "\n".join(
        [
            INPUT_PARSER_ERROR_HEADER_MESSAGE,
            "",
            "abcd",
            "  ^",
        ]
    )
